mongorestore/restore.go

```diff
@@ -306,17 +306,27 @@ func (restore *MongoRestore) RestoreIntent(intent *intents.Intent) Result {
 	return result
 }
 
-func (restore *MongoRestore) convertLegacyIndexes(indexes []IndexDocument, ns string) {
+func (restore *MongoRestore) convertLegacyIndexes(indexes []IndexDocument, ns string) []IndexDocument {
+	indexKeys := make(map[string]bool) // A set contains all the unique index's json strings
```

Comment:
small nit, but the go idiom is to use `map[string]struct{}` for sets

Reply:
Good idea!